ci-operator/step-registry/ibmcloud/deprovision/dedicatedhost/ibmcloud-deprovision-dedicatedhost-commands.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+function run_command() {
+    local CMD="$1"
+    echo "Running Command: ${CMD}"
+    eval "${CMD}"
+}
+
+# IBM Cloud CLI login
+function ibmcloud_login {
+    export IBMCLOUD_CLI=ibmcloud
+    export IBMCLOUD_HOME=/output
+    region="${LEASED_RESOURCE}"
+    export region
+    echo "Try to login..."
+    "${IBMCLOUD_CLI}" login -r ${region} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+}
+
+function deleteDedicatedHost() {
+    local dh=$1 dhg=$2 status
+    status=$(${IBMCLOUD_CLI} is dh $dh --output JSON | jq -r ."lifecycle_state")
+    if [[ "${status}" = "stable" ]]; then
+        run_command "${IBMCLOUD_CLI} is dhu ${dh} --enabled false"
+    fi
+
+    run_command "${IBMCLOUD_CLI} is dhd ${dh} -f"
+    run_command "${IBMCLOUD_CLI} is dhgd ${dhg} -f"
+}
+
+function run_command_with_retries() {
+  cmd="$1"
+  retries="$2"
+  interval="$3"
+
+  if [ X"$retries" == X"" ]; then
+      retries=20
+  fi
+
+  if [ X"$interval" == X"" ]; then
+      interval=30
+  fi
+
+  set +o errexit
+  output=$(eval "$cmd"); ret=$?
+  try=1
+
+  # avoid exit with "del Resource groups with active or pending reclamation instances can't be deleted"
+  while [ X"$ret" != X"0" ] && [ $try -lt $retries ]; do
+      sleep $interval
+      output=$(eval "$cmd"); ret=$?
+      try=$(expr $try + 1)
+  done
+  set -o errexit
+
+  if [ X"$try" == X"$retries" ]; then
+      return 2
+  fi
+  echo "$output"
+  return 0
+}
+
+ibmcloud_login
+
+dh_info="${SHARED_DIR}/dedicated_host"
+
+echo "DEBUG" "Removing the dedicated host based on ${dh_info}"
+dhg=$(cat "${SHARED_DIR}/dedicated_host_group")
+dh=$(cat "${SHARED_DIR}/dedicated_host")
+
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+rg=$(${IBMCLOUD_CLI} is dhg ${dhg} --output JSON | jq -r '.resource_group.name')
+
+echo "try to delete dedicated host ${dh} in dedicated host group ${dhg}"
+deleteDedicatedHost ${dh} ${dhg}
+
+if [[ ${rg} == "${CLUSTER_NAME}-rg" ]]; then
```

Comment:
What case need this if condition?

Reply:
rg (the resource group of the dedicated host group) is "Default", "${CLUSTER_NAME}-rg" will be deleted after uninstall; 
rg is "${CLUSTER_NAME}-rg", uninstall can not delete it, after the dh and dhg are all deleted, need delete the "${CLUSTER_NAME}-rg"

Reply:
> rg (the resource group of the dedicated host group) is "Default", "${CLUSTER_NAME}-rg" will be deleted after uninstall;

So in this case, "${CLUSTER_NAME}-rg" was created, but never use it, right?

Reply:
the cluster still installed in ${CLUSTER_NAME}-rg